### DISCUSSION TITLE
Is there any test framework used to write eclipse GUI testcases?

### DISCUSSION THREAD

**Discussion by lathapatil** [eclipse-platform/eclipse.platform.ui#1042] [Q&A] [answered]
I am planning to contribute couple of fixes and when trying to write Junit testcase and was checking if any Framework is used  for creating Eclipse GUI testcases which would save the time .
Unfortunately I did not find any like (SWT bot or Window Tester). I see the testcases are written using APIs of the code itself.

Please let me know if I am correct

**Comment by akurtakov** [comment, marked as answer]
Nothing in eclipse platform uses such gui testing libraries. In whole Eclipse TLP project there is only one plugin (https://github.com/eclipse-pde/eclipse.pde/tree/master/ui/org.eclipse.pde.ui.tests.smartimport) that uses [RedDeer](https://github.com/eclipse/reddeer).

**Comment by mickaelistria** [comment]
Indeed, Eclipse Platform code doesn't use much bots. There are issues with SWTBot that whenever the UI change, the test is likely to be broken (because SWTBot uses label, widget orders and other mutable things to locate the widget and interact; and also because it is all asynchronous while many tests in Platform require some finer synchronization work -wait for a build to finish, wait for something to render...).
RedDeer is using the page object framework which turn the tests very nice to develop and use and maintain but which make them more expensive to write in 1st place. So far, it didn't seem much profitable to use this pattern in most tests.
A benefit of using Platform API to write Platform tests is that is allows to validate the API and the test serve as a great resource of how APIs can and should be used. Using Platform API in Platform tests require no extra knowledge over developing Platform itself so the same people can be as efficient as writing tests vs "main" code. Usual Testing framework tend to serve the story of distinct teams doing main vs tests: the developer team write the functional code and low-level tests with low-level APIs they "own", while a Quality team write extra functional higher-level tests without need to know the deep details of the target application. Eclipse Platform doesn't have this split.

**Reply by lathapatil** [reply]
Thanks for the detailed explanation
